Library/Homebrew/dev-cmd/bottle.rb

```diff
@@ -439,7 +439,14 @@ def merge
     raise UsageError, "--merge requires a JSON file path argument" if Homebrew.args.named.blank?
 
     bottles_hash = Homebrew.args.named.reduce({}) do |hash, json_file|
-      hash.deep_merge(JSON.parse(IO.read(json_file)))
+      hash.deep_merge(JSON.parse(IO.read(json_file))) do |key, first, second|
+        if key == "cellar"
+          # Prioritize HOMEBREW_CELLAR over :any over :any_skip_relocation
+          [first, second].min
```

Comment:
Not sure I understand the `.min` here. Can you explain?

Reply:
It will return the minimum value from those 2 cellars.
`HOMEBREW_CELLAR` < `:any` < `:any_skip_relocation`

Reply:
Is that just based on a string comparison? If so it's a bit confusing and I think an `if`/`else` or a `case` statement might be a bit more obvious with intent.

Reply:
Yep, just based on string comparison. `/…` sorts before `any` sorts before `any_skip_relocation`. I'm personally okay with this style, so long as the comment precedes the code. The code to manually sort it could be a bit tedious. If we go that route, I'd suggesting creating a comparison function for this member and call `[first, second].min { |a, b| … }`

Reply:
@MikeMcQuaid Your call. What do you think?

Reply:
> The code to manually sort it could be a bit tedious.

Should only need three cases, right? Basically `HOMEBREW_CELLAR < :any < :any_skip_relocation` in logic form. I'd prefer that for explicitness. Can help with some suggested code if that would be useful.

Reply:
Please do.

Reply:
If I've understood correctly: how about: 

```suggestion
          cellars = [first, second]
          if cellars.include?(HOMEBREW_CELLAR)
            HOMEBREW_CELLAR
          elsif cellars.include?(:any)
            :any
          elsif cellars.include?(:any_skip_relocation)
            :any_skip_relocation
          else
            second
          end
```

Reply:
This code doesn't handle the case when the cellar is the default path from the other OS or a non default path. How about:
```ruby
if cellars.include?(HOMEBREW_CELLAR)
  HOMEBREW_CELLAR
elsif first[0] == "/"
  first
elsif second[0] == "/"
  second
elsif cellars.include?(:any)
  :any
else
  :any_skip_relocation
end
```

Reply:
Would probably want to see an `elsif cellars.include?(:any_skip_relocation)` partly just to ensure this code survives being changed in future (as an `else` it makes me a bit uneasy).

Reply:
Updated.